wolfcrypt/src/ecc.c

```diff
@@ -6380,12 +6385,15 @@ int wc_ecc_gen_deterministic_k(const byte* hash, word32 hashSz,
          * doing bits2octets(H(m)), when variant macro is used avoid this
          * bits2octets operation */
         if (mp_cmp(z1, order) == MP_GT) {
+            int z1Sz;
+
             mp_sub(z1, order, z1);
-            h1len = mp_unsigned_bin_size(z1);
-            if (h1len < 0 || h1len > WC_MAX_DIGEST_SIZE) {
+            z1Sz = mp_unsigned_bin_size(z1);
+            if (z1Sz < 0 || h1len > WC_MAX_DIGEST_SIZE) {
```

Comment:
hashSz is check against WC_MAX_DIGEST_SIZE above.
h1len is assigned to hashSz and not modified.